File: bench/perl/src/simple_id.pl
#! env perl
use strict;
use warnings;
use Time::HiRes qw/ time /;

# Perl variant of simple_id

my $regex = "(.*)";

my $pre_compile = time;

# Note the /s modifier.  This makes dot match newline.
my $pattern = qr/$regex/s;
my $lno = 0;

# Start timing
my $start = time;

while (<STDIN>) {
    $lno = $lno + 1;
    if ($_ =~ $pattern) {
        print STDOUT sprintf("%s", $1);
    } else {
        print STDERR "match error on line ${lno}\n";
        exit 1;
    }
}

# End timing
my $end = time;
my $elaps = int(($end - $start) * 1000);
my $elaps_compile = int(($start - $pre_compile) * 1000);

print STDERR "\ncompilation (ms): ${elaps_compile}\n";
print STDERR "matching (ms):    ${elaps}\n"

File: bench/python/src/simple_id.py
#! env python

# Python version of simple_id
import sys
import re
import datetime

regex = "(.*)"

pre_compile = datetime.datetime.now()

pattern = re.compile(regex, re.DOTALL)
lno = 0

# Start timing
start = datetime.datetime.now()

for line in sys.stdin:
    lno += 1
    m = pattern.match(line)
    if m:
        sys.stdout.write("%s" % m.group(0))
    else:
        sys.stderr.write("match error on line %s\n" % str(lno))
        exit(1)

# End timing
end = datetime.datetime.now()

# Elapsed time
elaps = end - start
elaps_compile = start - pre_compile
elaps_ms = elaps.seconds * 1000 + elaps.microseconds / 1000
elaps_compile_ms = elaps_compile.seconds * 1000 + elaps_compile.microseconds / 1000

sys.stderr.write("\ncompilation (ms): %s\n" % str(elaps_compile_ms))
sys.stderr.write("matching (ms):    %s\n" % str(elaps_ms))
